### PR TITLE
Add install and customize commands to VanillaSoftwareProcess

### PR DESCRIPTION
**Pull request by mikezaccardo**

### CODE REVIEW

File: software/base/src/main/java/brooklyn/entity/basic/VanillaSoftwareProcessSshDriver.java [hunk 1]
```diff
@@ -78,6 +80,16 @@ public void install() {
                     throw new IllegalStateException("Error installing archive: " + downloadedFilename);
             }
         }
+        else {
```
Comment (ahgittin) [inline software/base/src/main/java/brooklyn/entity/basic/VanillaSoftwareProcessSshDriver.java]: why 'else'?  i think always do this

File: software/base/src/main/java/brooklyn/entity/basic/VanillaSoftwareProcessSshDriver.java [hunk 2]
```diff
@@ -78,6 +80,16 @@ public void install() {
...
+            String installCommand = getEntity().getConfig(VanillaSoftwareProcess.INSTALL_COMMAND);
+            
+            if (!StringUtils.isBlank(installCommand)) {
```
Comment (ahgittin) [inline software/base/src/main/java/brooklyn/entity/basic/VanillaSoftwareProcessSshDriver.java]: prefer `Strings.isNonBlank(...)`

File: software/base/src/main/java/brooklyn/entity/basic/VanillaSoftwareProcessSshDriver.java [hunk 3]
```diff
@@ -90,6 +102,18 @@ public void customize() {
                     .body.append(ArchiveUtils.extractCommands(downloadedFilename, getInstallDir()))
                     .execute();
         }
+        else {
```
Comment (ahgittin) [inline software/base/src/main/java/brooklyn/entity/basic/VanillaSoftwareProcessSshDriver.java]: again, no `else`

**Comment by ahgittin** [comment]
minor comments, then looks good

**Comment by mikezaccardo** [comment]
Addressed

**Comment by ahgittin** [comment]
LGTM

File: software/base/src/main/java/brooklyn/entity/basic/VanillaSoftwareProcessSshDriver.java [hunk 4]
```diff
@@ -90,6 +100,17 @@ public void customize() {
                     .body.append(ArchiveUtils.extractCommands(downloadedFilename, getInstallDir()))
                     .execute();
         }
+        
+        String customizeCommand = getEntity().getConfig(VanillaSoftwareProcess.CUSTOMIZE_COMMAND);
+        
+        if (Strings.isNonBlank(customizeCommand)) {
+            newScript(CUSTOMIZING)
+                .failOnNonZeroResultCode()
+                // don't set vars yet -- it resolves dependencies (e.g. DB) which we don't want until we start
```
Comment (ahgittin) [inline software/base/src/main/java/brooklyn/entity/basic/VanillaSoftwareProcessSshDriver.java]: @mikezaccardo i'm finding vars can be useful in customize phase - going to reinstate this

**Pull request merged**